src/i18n/dictionaries.ts

```
@@ -0,0 +1,47 @@
+import 'server-only';
+
+import { getLocales } from '@/i18n/get-locales';
+
+export const getDictionary = async (locale: Locale) => {
+	const { locales } = await getLocales();
+
+	const dictionaries = locales.reduce(
+		(acc: Record<Locale, () => Promise<any>>, lang: Locale) => {
+			acc[lang] = () =>
+				import(`./dictionaries/${lang}.json`).then(
+					(module) => module.default
+				);
+			return acc;
+		},
+		{}
+	);
+
+	const dictionary = await dictionaries[locale]();
```

What happen if the wanted local does not exists? Does it completely crash or it silently fallback to the default one?

---

@kuuak For the moment it throws an error. But I will make it fallback to the default one, and show a warning message instead of throwing an error

---

Done !